packages/ui/aurora-grid/src/dnd/types.ts

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { ForwardRefExoticComponent, HTMLAttributes, RefAttributes } from 'react';
+
+// Definitions:
+// - Mosaic: container that manages the layout of a set of tiles (e.g., Stack, Grid, Tree, Table).
+// - Tile: component that is rendered within the layout and is draggable.
+// - Item: datum represented by a Tile.
+
+export type MosaicDataItem = { id: string };
+
+export type MosaicDraggedItem = {
+  container: string;
+  item: MosaicDataItem;
+  position?: any; // Index or layout-specific positional information (stored separately from the item).
+};
```

Comment:
You should be able to specify types for callbacks attached to specific containers

```suggestion
export type MosaicDraggedItem<TPosition = unknown> = {
  container: string;
  item: MosaicDataItem;
  position?: TPosition; // Index or layout-specific positional information (stored separately from the item).
};
```